### PR TITLE
pass auth info over fdb connections

### PR DESCRIPTION
**Pull request by mohitkhullar**
/plugin-branch fdb_iam

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Comment by riverszhang89** [comment]
Should we bump the FDB protocol version number for this feature? Remote may not understand the new addition `authdta`, even if source turns on the feature.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

File: plugins/remsql/fdb_comm.c [hunk 1]
```diff
@@ -1519,8 +1571,6 @@ static const char *__tran_2_str(enum transaction_level lvl)
         return "SOSQL";
     if (lvl == TRANLEVEL_RECOM)
         return "RECOM";
-    if (lvl == TRANLEVEL_MODSNAP)
-        return "MODSNAP";
```
Comment (riverszhang89) [inline plugins/remsql/fdb_comm.c:1523]: Did you mean to delete this?

**Comment by riverszhang89** [comment]
Passing the internal testsuite: `cdb2test   May 13 19:32:24 2024  in-comdb2db fdb_iam_fix.R20240513.5`

I've disabled the problematic test machine and resubmitted your PR.

File: plugins/remsql/fdb_comm.c (hunk 1 above)
Comment (mohitkhullar) [reply plugins/remsql/fdb_comm.c]: no, it was bad merge of recent changes. Thanks

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>11/533 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>truncatesc_ufid_on_generated<br>schemalk<br>schemalk_extralock_generated<br>phys_rep_tiered_nosource_generated<br>phys_rep_tiered<br>tunables<br>sc_downgrade<br>truncatesc_offline_generated<br>comdb2sys_queueodh_generated<br>comdb2sys_pagesize_generated</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>12/533 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>logfill<br>sc_resume<br>schemalk_extralock_generated<br>sc_transactional_rowlocks_generated<br>phys_rep_tiered_nosource_generated<br>phys_rep_tiered<br>sc_downgrade<br>truncatesc_offline_generated<br>selectv_rcode_serialretry_generated<br>comdb2sys_queueodh_generated</p>

**Review by riverszhang89 — APPROVED** [review]

**Comment by mohitkhullar** [comment]
Thanks

**Pull request merged**